services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -138,7 +141,22 @@ def compute_index_rows(
     logging.debug(CREATE_SEQUENCE_COMMAND)
     con.sql(CREATE_SEQUENCE_COMMAND)
 
-    create_command_sql = f"{CREATE_TABLE_COMMAND.format(columns=column_names)} read_parquet({parquet_urls});"
+    # see https://pypi.org/project/hf-transfer/ for more details about how to enable hf_transfer
+    os.environ["HF_HUB_ENABLE_HF_TRANSFER"] = "1"
+    for parquet_file in parquet_file_names:
+        hf_hub_download(
+            repo_type=REPO_TYPE,
+            revision=target_revision,
+            repo_id=dataset,
+            filename=f"{config}/{split}/{parquet_file}",
+            local_dir=duckdb_index_file_directory,
+            local_dir_use_symlinks=False,
+            token=hf_token,
+            cache_dir=duckdb_index_file_directory,
+        )
+
+    all_split_parquets = f"{duckdb_index_file_directory}/{config}/{split}/*.parquet"
+    create_command_sql = f"{CREATE_TABLE_COMMAND.format(columns=column_names)} '{all_split_parquets}';"
```

Comment:
We must delete the parquet files afterward, as split-descriptive-statistics does.

Reply:
Since this job runner inherits from SplitJobRunnerWithCache, they will be removed because `duckdb_index_file_directory` is the generated temporary `self.cache_subdirectory`.

Reply:
OK!!!